### PR TITLE
F32 && F64 tests

### PR DESCRIPTION
**Pull request by svyatonik**

### CODE REVIEW

File: src/interpreter/value.rs [hunk 1]
```diff
@@ -604,17 +606,40 @@ macro_rules! impl_float {
...
+			fn max(self, other: $type) -> $type {
+				if self.is_nan() || other.is_nan() {
+					use std::$type;
+					return $type::NAN;
+				}
+
+				self.max(other)
+			}
 			fn copysign(self, other: $type) -> $type {
 				// TODO: this may be buggy for edge cases
```
Comment (NikVolf) [inline src/interpreter/value.rs]: Still can be buggy after passing tests? :)
